### ISSUE TITLE
Custom interpolation?

### ISSUE THREAD

**Issue by JaneJeon** [pinojs/pino#1376]
Hi, searched the docs and the issues page for formatters and interpolation, but I don't see this anywhere.

Is there a way I can specify a custom interpolator (e.g. %t) akin to [this](https://github.com/debug-js/debug#formatters) so that `log.$method('%t', objToBeInterpolated)` can be interpolated in that custom format?

**Comment by mcollina** [comment]
I don't think this is supported but a PR to quick-format-unescaped  could add this capability.

**Comment by JaneJeon** [comment]
Okay, so full disclosure, I wanted the custom interpolator - %t - specifically to format milliseconds into a readable "datetime string"; however, it most certainly will require configuration (after all, there are as many ways to display time as there are milliseconds), and the current structure of quick-format-unescaped seems unfit to do something like this.

In particular, the fact that you're supposed to use the exported `format` function directly, and not have a generator that _returns_ the `format` function that will eventually be used, means that there is no space for configuration, which means you can't have custom interpolators in pinojs.

And if one were to change quick-format-unescaped to export a generator rather than the format function directly, it would most certainly be a breaking change, and I'm just not comfortable with breaking someone else's library, especially when it is being used by a large project like pino.

**Comment by mcollina** [comment]
> In particular, the fact that you're supposed to use the exported format function directly, and not have a generator that returns the format function that will eventually be used, means that there is no space for configuration, which means you can't have custom interpolators in pinojs.

> And if one were to change quick-format-unescaped to export a generator rather than the format function directly, it would most certainly be a breaking change/major semver, and I'm just not comfortable with breaking someone else's library, especially when it is being used by a large project like pino.

I'm pretty sure @davidmarkclements don't mind. I prefer that configuration pattern anyway, you can make it semver minor with the following trick:

```js
function build (...opts) {
  // do stuff
}
module.exports = build() // default options
module.exports.build = build
```

**Comment by JaneJeon** [comment]
I suppose so. But pino is using it, so it will have to adapt to this usage as well (or use that trick). Are you okay with that? If so, I’ll make the PR

**Comment by mcollina** [comment]
Yeah, no prob.

**Comment by JaneJeon** [comment]
Well, shit. I'm not able to make "the trick" work (the test complains that $import.build() is not a function, not to mention I'm preeeetty sure this trick doesn't work with ESM, too).

Would it be okay to just export build directly?

**Comment by JaneJeon** [comment]
Welp, published it anyway.
